app/src/modules/settings/routes/marketplace/routes/registry/registry.vue

```diff
@@ -0,0 +1,197 @@
+<script setup lang="ts">
+import api from '@/api';
+import VBanner from '@/components/v-banner.vue';
+import type { RegistryListResponse } from '@directus/extensions-registry';
+import { useRouteQuery } from '@vueuse/router';
+import { isEqual } from 'lodash';
+import { computed, ref, watch, watchEffect } from 'vue';
+import { useI18n } from 'vue-i18n';
+import SettingsNavigation from '../../../../components/navigation.vue';
+import ExtensionListItem from '../../components/extension-list-item.vue';
+import InlineFilter from './components/inline-filter.vue';
+import RegistryInfoSidebarDetail from './components/registry-info-sidebar-detail.vue';
+
+const { t } = useI18n();
+
+const perPage = 10;
+
+const search = useRouteQuery('search');
+const page = useRouteQuery('page', 1);
+const type = useRouteQuery('type');
+const sort = useRouteQuery<'popular' | 'recent' | 'downloads'>('sort', 'popular');
```

Comment:
Having state in the URL is really nice now, but the initial problem of not being able to properly use the browsers back button remains.

1. Go to settings
2. Go to marketplace
3. Press onto some pagination page button, lets say page 3
4. Press the back button
5. You will now go back to the settings instead of going back to the first market place page

This is because we dont push the new URL to the page history

Reply:
Curse and a blessing? 🤔 It would also be super annoying to have everything in the history state cause then you can never click back to get back to the previous page

Reply:
I look at it differently 😬 And do consider each page in "pagination" to be its own page (or more broadly each unique filter combination of `type`,`page`,`sort`)

> It would also be super annoying to have everything in the history state cause then you can never click back to get back to the previous page

In its current state it is impossible to get back to the previous page because it does not exist in the history while when everything is on the browser history it is a question of going back enough times to get where you wanted to be (it can take a while if you've done a lot of browsing but at least you can get back if wanted).

Reply:
FWIW: I'm in the same camp as @br41nslug - Personally I'd strongly prefer it being in the history :+1:

Reply:
Can we push stuff in history in Vue without triggering a navigation event (eg reload)? If that's possible I'm willing to concede.

Reply:
I think a direct [`history.pushState`](https://developer.mozilla.org/en-US/docs/Web/API/History/pushState) call should do it.

Reply:
Spend about an hour or so trying to sync it into `history.pushState` directly, but am getting some very weird behavior. I'm assuming it's because of vue-router and the browser history getting out of sync while manually adjusting the browser history state. Using `vue-router` `route.push` directly doesn't seem to actually push it into the history state. I'm assuming it's because the vue-router is already on the 'current' route when the change happens. It won't add an entry if it's already the page we're on.

Long story short, I'm treating it as a nice to have but not a blocker for this PR. Lets plan on circling back to this later 👍🏻